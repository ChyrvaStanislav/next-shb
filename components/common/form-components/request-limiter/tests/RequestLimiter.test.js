import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import RequestLimiter from '../index';

const onFormClose = jest.fn();

const Component = props => <RequestLimiter {...props} onFormClose={onFormClose} />;

describe('RequestLimiter renders', () => {
  it('should RequestLimiter renders', () => {
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should RequestLimiter renders with type', () => {
    const { asFragment } = render(<Component type="subscribe" />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('RequestLimiter functions', () => {
  it('should onFormClose works', () => {
    const { getByText } = render(<Component />);
    const button = getByText('Continue Exploring');
    fireEvent.click(button);

    expect(onFormClose).toBeCalled();
  });
});
