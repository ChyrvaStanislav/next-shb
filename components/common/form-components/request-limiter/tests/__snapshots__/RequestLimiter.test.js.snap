// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestLimiter renders should RequestLimiter renders 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <div
      class="requestLimiterIcon"
    >
      <span />
      <span />
    </div>
    <p
      class="requestLimiterHeading"
    >
      You have reached our daily limit for submit actions
    </p>
    <p
      class="requestLimiterDescription"
    >
      Try to submit your request next day.
    </p>
    <button
      class="button primary medium requestLimiterButton"
      type="submit"
    >
      Continue Exploring
    </button>
  </div>
</DocumentFragment>
`;

exports[`RequestLimiter renders should RequestLimiter renders with type 1`] = `
<DocumentFragment>
  <div
    class="root rootSubscribe"
  >
    <div
      class="requestLimiterIcon"
    >
      <span />
      <span />
    </div>
    <p
      class="requestLimiterHeading"
    >
      You have reached our daily limit for submit actions
    </p>
    <p
      class="requestLimiterDescription"
    >
      Try to submit your request next day.
    </p>
  </div>
</DocumentFragment>
`;
